### PR TITLE
Correct files included in a source distribution

### PR DESCRIPTION
**Pull request by phargogh**
This PR updates `MANIFEST.in` to better-restrict the files that are include in a source distribution.

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -2,7 +2,5 @@ include HISTORY.rst
 include LICENSE.txt
 include README.rst
 include pyproject.toml
-global-include *.pyx
-global-include *.pxd
-global-include *.h
 include requirements.txt
+recursive-include src *.h *.cpp *.pyx *.pxd
```
Comment (dcdenu4) [inline MANIFEST.in:6]: OK, so this is just including the files with stated extensions from src. Seems like a good idea.

File: HISTORY.rst [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
-..
-  Unreleased Changes
-  -----------------
+
+Unreleased Changes
+------------------
+* Updated ``MANIFEST.in`` to only include files that should be there in the
+  pygeoprocessing source distribution.  This fixes an issue where files
+  matching a variety of extensions anywhere in the pygeoprocessing directory
+  might be included with the source distribution.
```
Comment (dcdenu4) [inline HISTORY.rst:10]: Ah, this answers and confirms my comment. Also, multiline comments!

**Review by dcdenu4 — APPROVED** [review]
Looks good to me! Good catch on this James. Assuming test checks pass of course!

File: HISTORY.rst (hunk 2 above)
Comment (phargogh) [reply HISTORY.rst:10]: I know, right??  Super helpful.

File: MANIFEST.in (hunk 1 above)
Comment (phargogh) [reply MANIFEST.in:6]: Yep, that's the idea!  I never thought that `global-include` would reach inside a virtual environment, but there you have it.

**Review by davemfish — APPROVED** [review]

**Pull request merged**